doc/side-by-side-CTL-and-Plutus-comparison.md

```diff
@@ -0,0 +1,323 @@
+-- | `mustPayToScript`, and all scripts must be explicitly provided to build
+-- | the transaction.
+mustPayToScript
+  :: forall (i :: Type) (o :: Type)
+   . ValidatorHash
+  -> Datum
+  -> Value
+  -> TxConstraints i o
+```
+
+### The `give` contract 
+
+Now we can write and compare the `give` contract. 
+This contract takes an amount of Ada from our wallet 
+an lock it behind the script that validates any
+transactions.
+
+```Haskell
+-- Haskell 
+give :: 
+  forall (w::Type) (s::Type) (e::Type) . 
+    AsContractError e => 
+      Integer -> Contract w s e ()
+give amount = do
+   let tx = mustPayToOtherScript vHash (Datum $ Constr 0 []) 
+              $ Ada.lovelaceValueOf amount
+   ledgerTx <- submitTx tx
+   void $ awaitTxConfirmed $ txId ledgerTx
+   logInfo @String $ printf "made a gift of %d lovelace" amount
+```
+
+We include some of the imports for the PureScript 
+contract. 
+
+```PureScript
+-- PureScript
+import Contract.PlutusData (PlutusData, unitDatum)
+import Contract.ScriptLookups as Lookups
+import Contract.TxConstraints as Constraints
+import Contract.Prelude
+import Data.BigInt as BigInt
+
+give :: ValidatorHash -> Validator -> Contract () TransactionHash
+give vhash validator = do
+  let
+    constraints :: Constraints.TxConstraints Unit Unit
+    constraints = Constraints.mustPayToScript vhash unitDatum
+      $ Value.lovelaceValueOf
+      $ BigInt.fromInt 2_000_000
+
+    lookups :: Lookups.ScriptLookups PlutusData
+    lookups = Lookups.validator validator
+
+  buildBalanceSignAndSubmitTx lookups constraints
+```
+
+
+Notice how with the current API offered by CTL we need to provide the
+validator as part of the `lookups` by ourselves. 
+
+
+### The `grab` contract
+
+The Plutus `grab` example takes all the UTxOs locked by 
+the on-chain contract that always validates a transaction, and spends 
+them to get all in the wallet of the user running the example.
+This isn't a problem as the example is intended to run inside a 
+Plutus `EmulatorTrace` in a local toy environment.
+
+```Haskell
+-- Haskell
+grab :: forall w s e. AsContractError e => Contract w s e ()
+grab = do
+   utxos <- utxoAt scrAddress
+   let orefs   = fst <$> Map.toList utxos
+      lookups  = Constraints.unspentOutputs utxos      <>
+                 Constraints.otherScript validator
+      tx :: TxConstraints Void Void
+      tx       = mconcat [mustSpendScriptOutput oref $ Redeemer $ I 17 | oref <- orefs]
+   ledgerTx <- submitTxConstraintsWith @Void lookups tx
+   void $ awaitTxConfirmed $ txId ledgerTx
+   logInfo @String $ "collected gifts"
+```
+
+To talk about the grab contract in CTL we need to talk about some 
+functions and types of CTL first. 
+
+```PureScript
+module Plutus.Types.Transaction ... 
+.
+.
+.
+type Utxo = Map TransactionInput TransactionOutput
+newtype UtxoM = UtxoM Utxo
+```
+
+```PureScript
+module Contract.Utxos ...
+.
+.
+.
+-- | This module defines query functionality via Ogmios to get utxos. Gets 
```

Comment:
Please use `UTxO` instead of `utxo` spelling in docs and comments.

I will do the same change to existing docs soon.